apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -228,19 +229,49 @@ namespace MWRender
         resourceSystem->getSceneManager()->setAutoUseSpecularMaps(Settings::Manager::getBool("auto use object specular maps", "Shaders"));
         resourceSystem->getSceneManager()->setSpecularMapPattern(Settings::Manager::getString("specular map pattern", "Shaders"));
 
+        osg::ref_ptr<osg::Group> scenegroup=new osg::Group;
         osg::ref_ptr<SceneUtil::LightManager> sceneRoot = new SceneUtil::LightManager;
+        mSceneRoot = scenegroup;
         sceneRoot->setLightingMask(Mask_Lighting);
-        mSceneRoot = sceneRoot;
+
+        mRootNode->addChild(sceneRoot);
+
+        mSky.reset(new SkyManager(sceneRoot, resourceSystem->getSceneManager()));
+         osg::LightSource *sunlightsource = new osg::LightSource;
+            mSunLight = new osg::Light;
+            sunlightsource->setLight(mSunLight.get());
+            mSunLight->setDiffuse(osg::Vec4f(0,0,0,1));
+            mSunLight->setAmbient(osg::Vec4f(0,0,0,1));
+            mSunLight->setSpecular(osg::Vec4f(0,0,0,0));
+            mSunLight->setConstantAttenuation(1.f);
+            sunlightsource->setNodeMask(Mask_Lighting);
+            sceneRoot->addChild(sunlightsource);
+        sunlightsource->setStateSetModes(*mRootNode->getOrCreateStateSet(), osg::StateAttribute::ON);
+
+
+        sceneRoot->getOrCreateStateSet()->setMode(GL_CULL_FACE, osg::StateAttribute::ON);
+        sceneRoot->getOrCreateStateSet()->setMode(GL_LIGHTING, osg::StateAttribute::ON);
+        sceneRoot->getOrCreateStateSet()->setMode(GL_NORMALIZE, osg::StateAttribute::ON);
+        osg::ref_ptr<osg::Material> defaultMat (new osg::Material);
+        defaultMat->setColorMode(osg::Material::OFF);
+        defaultMat->setAmbient(osg::Material::FRONT_AND_BACK, osg::Vec4f(1,1,1,1));
+        defaultMat->setDiffuse(osg::Material::FRONT_AND_BACK, osg::Vec4f(1,1,1,1));
+        defaultMat->setSpecular(osg::Material::FRONT_AND_BACK, osg::Vec4f(0.f, 0.f, 0.f, 0.f));
+        sceneRoot->getOrCreateStateSet()->setAttribute(defaultMat);
+
+        scenegroup->setNodeMask(Mask_Scene);
+        scenegroup->setName("Scene Root");
```

Comment:
You've still not explained what these changes do, how they do it, or why it's necessary.

Reply:
just add a derivation to the tree in order to choose weither or not i cast shadow on

Reply:
That only partially answers the first of those three questions.

Reply:
arg why did all my comment have disappeared?
Have you had the time to read the answer i wrote here?